contributor-docs/operations/tests/rfc2119-keyword-test-guide.md

```diff
@@ -0,0 +1,157 @@
+
+- Operating System: The server **MUST** run on an OS that receives current security updates.
+  Unsupported versions **MUST NOT** be used.
+
+- TLS Configuration: TLS 1.2 or higher **MUST** be enabled for all connections. SSL and
+  earlier TLS versions **MUST NOT** be used.
+
+- Security Patches: The web server and OS **MUST** be kept up-to-date with security patches.
+  Critical patches **SHOULD** be applied within 48 hours of release.
+
+- Firewall Setup: The firewall **SHOULD** be configured to allow only necessary ports,
+  such as 80 (HTTP) and 443 (HTTPS). All unnecessary ports **SHOULD NOT** be open.
+
+- Backup Plan: Regular backups of the web server data **MAY** be performed, and it is
+  recommended that backup integrity checks are conducted to ensure recoverability.
+
+#### Compliance
+
+Failure to comply with these requirements results in the web server being
+considered non-compliant with the organization's security policies.
+
+## 6. Example: Test script using RFC2119 Keywords
+
+The following example of a fictitious code for a test of "process_requirements"
+is intended to illustrate the use of RFC2119 keywords in connection with the
+creation of SCS-compliant standard tests.
+
+```python
+
+import logging
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+def process_requirements(var1, var2):
+    try:
+        # debug log for input variables
+        logger.debug(f"Received input var1: {var1}, var2: {var2}")
+
+        # example of an error case (equivalent to MUST and MUST NOT)
+        if var1 < 0:
+            logger.error("var1 must be positive")
+        elif var1 > 100:
+            logger.error("var1 must not exceed 100")
+
+        # add debug log to confirm evaluation of var1
+        logger.debug(f"var1 evaluated: {var1 > 0 and var1 < 100}")
+
+        # example of a warning case (equivalent to SHOULD and SHOULD NOT)
+        if var2 < 10:
+            logger.warning("var2 should be at least 10 for optimal performance")
+        elif var2 > 50:
+            logger.warning("var2 should not exceed 50 for optimal performance")
+
+        # add debug log to confirm evaluation of var2
+        logger.debug(f"var2 evaluated: {var2 > 10 and var2 < 50}")
+
+        # example of an informative case (equivalent to MAY)
+        if var1 + var2 < 100:
+            logger.info("Combination is within recommended range")
+        else:
+            logger.info("Combination exceeds the recommended range but may still proceed")
+
+        # debug log before completing the process
+        logger.debug(f"Final evaluation before completion: var1 + var2 = {var1 + var2}")
+
+        # success log
+        logger.debug("Process completed successfully")
+
+    except ValueError as e:
+        # debug log before raising a critical error
+        logger.debug("Encountered a critical error about to log", exc_info=True)
+        logger.critical(f"Process failed: {e}", exc_info=True)
+
+```
```

Comment:
IMHO somewhere in this document it should be documented that the exit code of the conformance test script must be non-zero if we have ERROR or CRITICAL messages.

Reply:
I added this as well.